src/uffd_handler/src/memory_region.rs

```diff
@@ -119,37 +173,73 @@ mod tests {
         ];
 
         let msg = serde_json::to_string(&expected_mappings).unwrap();
-        let mappings = deserialize_mappings(&msg, PAGE_SIZE * 5).unwrap();
+        let mappings = deserialize_mappings(&msg).unwrap();
+        assert!(mappings_sanity_checks(&mappings, PAGE_SIZE * 5).is_ok());
         assert_eq!(mappings, expected_mappings);
     }
 
     #[test]
-    fn test_deserialize_mappings_corrupted() {
-        let mappings = [GuestRegionUffdMapping {
+    fn test_deserialize_mappings_invalid() {
+        let res = deserialize_mappings("foo bar");
+        assert!(res.is_err());
+        assert!(matches!(
+            res.err().unwrap(),
+            Error::DeserializeMemoryMappings(_)
+        ));
+
+        let res = deserialize_mappings("");
+        assert!(res.is_err());
+        assert!(matches!(
+            res.err().unwrap(),
+            Error::DeserializeMemoryMappings(_)
+        ));
+    }
+
+    #[test]
+    fn test_mappings_sanity_checks_corrupted() {
+        let mut mappings = vec![GuestRegionUffdMapping {
             base_host_virt_addr: 0x1000,
             size: PAGE_SIZE,
             offset: 0,
         }];
-        let expected_size = PAGE_SIZE - 1;
-        let msg = serde_json::to_string(&mappings).unwrap();
 
-        // Deserialization fails because mappings don't match the expected size,
-        // thus they are corrupted.
-        let res = deserialize_mappings(&msg, expected_size);
+        // Sanity checks fail because mappings size doesn't match the expected file size.
+        let res = mappings_sanity_checks(&mappings, PAGE_SIZE + 1);
         assert!(res.is_err());
         assert_eq!(
-            Error::CorruptedMemoryMappings.to_string(),
+            MemRegionError::SizeMismatch.to_string(),
             res.err().unwrap().to_string()
         );
-    }
 
-    #[test]
-    fn test_deserialize_mappings_invalid() {
-        let res = deserialize_mappings("foo bar", 0);
+        // Sanity checks fail because the region does not fit into the file entirely.
+        // The region begins at offset=0x100 and ends at 0x1100, while the file's size is
+        // PAGE_SIZE=0x1000.
+        mappings[0].offset = 0x100;
+        let res = mappings_sanity_checks(&mappings, PAGE_SIZE);
         assert!(res.is_err());
-        assert!(matches!(
-            res.err().unwrap(),
-            Error::DeserializeMemoryMappings(_)
-        ));
+        assert_eq!(
+            MemRegionError::MappingPastEof.to_string(),
+            res.err().unwrap().to_string()
+        );
+
+        // Sanity checks fail because the specified file offset and length
+        // cause overflow when added.
+        mappings[0].offset = u64::MAX;
+        let res = mappings_sanity_checks(&mappings, PAGE_SIZE);
+        assert!(res.is_err());
+        assert_eq!(
+            MemRegionError::InvalidOffsetLength.to_string(),
+            res.err().unwrap().to_string()
+        );
```

Comment:
nit: I would do 
```suggestion
        assert_eq!(res, Err(MemRegionError::InvalidOffsetLength));
```
or if the `PartialEq` bounds are not satisfied, `assert!(matches!(...))` (same thing in a lot of other places).

Reply:
Done!

Reply:
ah, I meant `assert!(matches!(res, Err(MemRegionError::InvalidOffsetLength))`, sorry, wasn't clear. Anyway, no need for you to go through the PR again, its just a nit :)

(fyi, you can do `res.unwrap_err()` instead of `res.err().unwrap()`)